syntaxes/cshell.tmLanguage.json

```
@@ -20,15 +20,19 @@
     "keywords": {
       "patterns": [
         {
-          "match": "(?<=^|;|&|\\s)(if|elif|while|until)(?=\\()",
+          "match": "(?<=^|;|&|\\s)(if|else if|switch|while)(?= *\\(.*\\))",
           "name": "keyword.control.shell"
         },
         {
-          "match": "(?<=^|;|&|\\s)(setenv|@)(?=\\s|;|&|$)",
+          "match": "(?<=^|;|&|\\s)(set|setenv|@)(?=\\s|;|&|$)",
           "name": "storage.modifier.shell"
         },
         {
-          "match": "(?<=^|;|&|\\s)(switch|end|endif|endsw|foreach)(?=\\s|;|&|$)",
+          "match": "(?<=^|;|&|\\s)(else|end|endif|then)(?=\\s|;|&|$)",
```

```suggestion
          "match": "(?<=^|;|&|\\s)(switch|end|endif|endsw|foreach)(?=\\s|;|&|$)",
```

`else` and `then` is handled internally by VSCode by default.

---

Ok, my misunderstanding. See #2.